### PR TITLE
[Eslint]: fix errors according to eslint configuration

### PR DESCRIPTION
**Pull request by Sushasgit**
- [X] Lint and unit tests pass locally with my changes

### Description:
Fix issues according to eslint configuration for the following rules:
- @typescript-eslint/no-redeclare
- @typescript-eslint/no-shadow
- @typescript-eslint/no-unused-expressions
- @typescript-eslint/no-unused-vars
- jsx-a11y/img-redundant-alt
-  no-var
- eqeqeq

Regarding `no-unused-vars` added `// eslint-disable-next-line @typescript-eslint/no-unused-vars` to data source - this is a temporary solution - because from the first look it'll be hard to fix and be sure that everything is still working. Files:
- uui-core/src/data/processing/LazyDataSource.tsx
- uui-core/src/types/dataSources.ts
- uui-components/src/layout/flexItems/FlexSpacer.tsx (props are using as const priority = child.props.priority || index;) so we can't just delete typings.

### CODE REVIEW

File: uui-components/src/navigation/MainMenu/Burger/Burger.tsx [hunk 1]
```diff
@@ -99,9 +99,9 @@ export class Burger extends React.Component<BurgerProps, BurgerState> {
                     className={ cx(this.props.cx, uuiBurger.menu, css.container, this.state.isOpen && uuiBurger.menuOpen) }
                     { ...this.props.rawProps }
                 >
-                    <div className={ uuiBurger.button } onClick={ this.toggleBurgerMenu }>
+                    <button className={ uuiBurger.button } onClick={ this.toggleBurgerMenu }>
```
Comment (AlekseyManetov) [inline uui-components/src/navigation/MainMenu/Burger/Burger.tsx:102]: Do we need to reset default styles from button here?

File: uui-components/src/table/DataTableRow.tsx [hunk 2]
```diff
@@ -62,7 +62,7 @@ const DataTableRowImpl = React.forwardRef(function DataTableRow<TItem, TId>(prop
                 rawProps={ {
                     ...params.eventHandlers,
                     role: 'row',
-                    'aria-expanded': props.isFolded == undefined ? undefined : !props.isFolded,
+                    'aria-expanded': props.isFolded === undefined ? undefined : !props.isFolded,
```
Comment (AlekseyManetov) [inline uui-components/src/table/DataTableRow.tsx:65]: i guess here we can also have isFolded null value, so let's add this case with null also.

File: uui-core/src/hooks/useTableState/useTableState.ts [hunk 3]
```diff
@@ -179,15 +179,15 @@ export const useTableState = <TFilter = Record<string, any>, TViewState = any>(p
 
     const createNewPreset = useCallback(
         (name: string) => {
-            const tableStateValue = getTableStateValue();
+            const newTableStateValue = getTableStateValue();
```
Comment (AlekseyManetov) [inline uui-core/src/hooks/useTableState/useTableState.ts:182]: this is not new state, let's rename to currentTableStateValue

File: uui-core/src/hooks/useTableState/useTableState.ts [hunk 4]
```diff
@@ -199,12 +199,12 @@ export const useTableState = <TFilter = Record<string, any>, TViewState = any>(p
 
     const hasPresetChanged = useCallback(
         (preset: ITablePreset<TFilter, TViewState>) => {
-            const tableStateValue = getTableStateValue();
+            const newTableStateValue = getTableStateValue();
```
Comment (AlekseyManetov) [inline uui-core/src/hooks/useTableState/useTableState.ts:202]: this is not new state, let's rename to currentTableStateValue

**Review by AlekseyManetov — CHANGES_REQUESTED** [review]

File: uui-components/src/navigation/MainMenu/Burger/Burger.tsx (hunk 1 above)
Comment (Sushasgit) [reply uui-components/src/navigation/MainMenu/Burger/Burger.tsx:102]: Yeah, thanks fixed.

**Pull request merged**